src/js/tabs/register.js

```diff
@@ -137,39 +137,46 @@ RegisterTab.prototype.angular = function (module) {
       });
     };
 
+    var _resnedEmail = function(masterkey) {
+      id.resendEmail({
+        id:$scope.userBlob.id,
+        url:$scope.userBlob.url,
+        username: $scope.userCredentials.username,
+        account_id: $scope.userBlob.data.account_id,
+        email: $scope.newEmail || $scope.userBlob.data.email,
+        masterkey: masterkey
+      }, function(err, response){
+        if (err) {
+          console.log('Error', err);
+          return;
+        }
+
+        // Update the blob
+        $scope.userBlob.set('/email', $scope.newEmail || $scope.userBlob.data.email);
+
+        $scope.resendLoading = false;
+        $scope.resendSuccess = true;
+      });
+    }
+
     $scope.resendEmail = function()
     {
       $scope.resendLoading = true;
 
-      keychain.requestSecret(id.account, id.username,
-        function (err, masterkey) {
-          if (err) {
-            console.log('client: register tab: error while ' +
-              'unlocking wallet: ', err);
-            $scope.mode = 'verification';
-            return;
-          }
-
-          id.resendEmail({
-            id:$scope.userBlob.id,
-            url:$scope.userBlob.url,
-            username: $scope.userCredentials.username,
-            account_id: $scope.userBlob.data.account_id,
-            email: $scope.newEmail || $scope.userBlob.data.email,
-            masterkey: masterkey
-          }, function(err, response){
+      if ($scope.keyOpen) {
+        _resnedEmail($scope.keyOpen);
```

Comment:
If typo "happens" 3 times in a row, either you mass-rename the parameter, or you used auto-complete to type this.